### PR TITLE
Host List Expand All and Survey List -> Tables

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
This PR addresses #10654 #10739 and #10740.  The biggest change here is the way the use will edit the Survey question order.  We wanted to add drag functionality and @trahman73 decided that the best way to do it was to repurpose the Preview modal(which had the same data on it as the list itself) to be a order edit modal that allows the user to drag their questions where they want.  This also means that the user does not have to wait for the UI to POST the new order and fetch the new order from the API every time they change a single question.  They can do all of there changes at 1 time and then save those changes.
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - API
 - UI

##### ADDITIONAL INFORMATION
host expand
SurveyChanges

### CODE REVIEW

File: awx/ui/src/screens/Template/Survey/SurveyList.js [hunk 1]
```diff
@@ -129,36 +104,47 @@ function SurveyList({
...
-            question={question}
-            isChecked={selected.some((q) => q.variable === question.variable)}
-            onSelect={() => handleSelect(question)}
-            onMoveUp={moveUp}
-            onMoveDown={moveDown}
-            canEdit={canEdit}
-          />
-        ))}
+      <>
+        <TableComposable>
```
Comment (nixocio) [inline awx/ui/src/screens/Template/Survey/SurveyList.js]: We could add datalabel, ids, or ouiaId to help identifying all those table elements. Ease testing.

**Review by nixocio — CHANGES_REQUESTED** [review]
This is a nice UX improvement.

A few suggestions.

1. Add a note about drag and drop here.
image

Something like:

image

2. Disable Edit Order button if there is just one survey

image

image

3. Add a tooltip to `Edit Order` button.
4. Add a `Answer variable name` `blah` and use the drag and drop - click on the survey, and it will redirect away from the app.

**Review by nixocio — APPROVED** [review]
This a great UX improvement. Good job @AlexSCorey.

**Comment by akus062381** [comment]
@AlexSCorey

Would you please add better locators (OUIAID, data-cy, ID) to the expanded list items in the Host table, as shown below?
image

Can you please add better locators for the places I have circled?
image

Thank you!

**Review by akus062381 — APPROVED** [review]

**Comment by nixocio** [comment]
Since there were more change since the last time we ran the e2e tests, I will apply this label again.

**Pull request merged**